src/language/mongoDBService.ts

```diff
@@ -557,6 +595,86 @@ export default class MongoDBService {
     return [];
   }
 
+  // Highlight the usage of commands that only works inside interactive session.
+  provideDiagnostics(textFromEditor: string) {
+    const lines = textFromEditor.split(/\r?\n/g);
+    const diagnostics: Diagnostic[] = [];
+    const invalidInteractiveSyntaxes = [
+      { issue: 'use ', fix: "use('database')" },
+      { issue: 'show databases', fix: 'db.getMongo().getDBs()' },
+      { issue: 'show dbs', fix: 'db.getMongo().getDBs()' },
+      { issue: 'show collections', fix: 'db.getCollectionNames()' },
+      { issue: 'show tables', fix: 'db.getCollectionNames()' },
+      {
+        issue: 'show profile',
+        fix: "db.getCollection('system.profile').find()",
+      },
+      { issue: 'show users', fix: 'db.getUsers()' },
+      { issue: 'show roles', fix: 'db.getRoles({ showBuiltinRoles: true })' },
+      { issue: 'show logs', fix: "db.adminCommand({ getLog: '*' })" },
+      { issue: 'show log ', fix: "db.adminCommand({ getLog: 'global' })" },
+    ];
+
+    lines.forEach((line, i) => {
+      invalidInteractiveSyntaxes.forEach(({ issue, fix }) => {
+        const startCharacter = line.indexOf(issue);
+        let endCharacter = startCharacter + issue.length;
+
+        if (startCharacter >= 0) {
+          if (issue === 'use ') {
+            let databaseName = 'database';
+
+            this._cachedDatabases.forEach((item: CompletionItem) => {
+              if (line.indexOf(`use ${item.label}`) >= 0) {
```

Comment:
I’m wondering if it makes more sense to validate these items more strictly. For example, right now, I think this would also catch cases such as `print("I want a new mouse in my house")`, right?

I think it probably makes sense to at least check that there’s nothing but whitespace before the `use` keyword, and it probably also makes sense to ensure that the check here returns the same `use` keywords as the one above, e.g.

```suggestion
              if (line.substring(startCharacter).startsWith(`use ${item.label}`)) {
```